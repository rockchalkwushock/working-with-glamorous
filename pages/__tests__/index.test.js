import { shallow, render, mount } from 'enzyme'

import { TestPage } from '../'

describe('TestPage', () => {
  let ui
  test('should render to DOM', () => {
    ui = <TestPage />
    expect(shallow(ui)).toMatchSnapshotWithGlamor(`enzyme.shallow`)
    expect(mount(ui)).toMatchSnapshotWithGlamor(`enzyme.mount`)
    expect(render(ui)).toMatchSnapshotWithGlamor(`enzyme.render`)
  })

  test('should have initial state: false', () => {
    ui = shallow(<TestPage />)
    expect(ui.state('view')).toEqual(false)
  })

  test('should have state: true after onClick event', () => {
    ui = shallow(<TestPage />)
    expect(ui.state('view')).toEqual(false)
    ui.instance()._onClick() // this is so awesome, enzyme === super boss!!!
    expect(ui.state('view')).toEqual(true)
  })

  test('should revert to initial state upon second onClick event', () => {
    ui = shallow(<TestPage />)
    expect(ui.state('view')).toEqual(false)
    ui.instance()._onClick()
    expect(ui.state('view')).toEqual(true)
    ui.instance()._onClick()
    expect(ui.state('view')).toEqual(false)
  })
})
