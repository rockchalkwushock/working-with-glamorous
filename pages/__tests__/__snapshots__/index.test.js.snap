// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme.mount 1`] = `
.css-q44vm6,
[data-css-q44vm6] {
  -webkit-transition: all 0.3s ease-in-out;
  -webkit-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  top: 50%;
  left: 50%;
  height: 70px;
  position: absolute;
  width: 70px;
  border-radius: 6px;
  background-color: rgba(255,255,255,0);
  transition: all 0.3s ease-in-out;
  display: block;
  overflow: hidden;
}

<TestPage>
  <SearchBar
    onClick={[Function]}
    view={false}
  >
    <StyledWrapper
      onClick={[Function]}
      open={false}
    >
      <div
        className="css-q44vm6"
        onClick={[Function]}
      />
    </StyledWrapper>
  </SearchBar>
</TestPage>
`;

exports[`enzyme.render 1`] = `
.css-q44vm6,
[data-css-q44vm6] {
  -webkit-transition: all 0.3s ease-in-out;
  -webkit-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  top: 50%;
  left: 50%;
  height: 70px;
  position: absolute;
  width: 70px;
  border-radius: 6px;
  background-color: rgba(255,255,255,0);
  transition: all 0.3s ease-in-out;
  display: block;
  overflow: hidden;
}

<div
  class="css-q44vm6"
/>
`;

exports[`enzyme.shallow 1`] = `
<SearchBar
  onClick={[Function]}
  view={false}
/>
`;
